ci-operator/step-registry/sandboxed-containers-operator/gather-must-gather/sandboxed-containers-operator-gather-must-gather-commands.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "=========================================="
+echo "Sandboxed Containers Operator - Optional Must-Gather"
+echo "=========================================="
+
+# Check if must-gather is enabled
+if [[ "${ENABLE_MUST_GATHER:-true}" != "true" ]]; then
+    echo "Must-gather collection is disabled (ENABLE_MUST_GATHER=${ENABLE_MUST_GATHER:-})"
+    echo "Skipping must-gather step"
+    exit 0
+fi
+
+# Check if we should only run on failure
+if [[ "${MUST_GATHER_ON_FAILURE_ONLY:-false}" == "true" ]]; then
+    echo "Must-gather configured to run only on test failures"
+
+    # Check for test failure indicators
+    test_failed=false
+
+    # Method 1: Check for JUnit test results with failures
+    if find "${ARTIFACT_DIR}" -name "junit_*.xml" -type f 2>/dev/null | head -1 | xargs grep -q 'failures="[1-9]' 2>/dev/null; then
```

Comment:
I find this `find` too fragile, to be honest.

For example it will find [junit_operator.xml](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/69558/rehearse-69558-periodic-ci-openshift-sandboxed-containers-operator-devel-downstream-release-azure-ipi-peerpods/1970629233713614848/artifacts/junit_operator.xml) which contains the list of steps and their status (pass/fail). Is this file what you are looking for?

My point: we can be more specific about the files we are looking for, the better because we will avoid false-positives (and negatives).